internal/transport/http2_server.go

```diff
@@ -404,6 +411,25 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 		}
 	}
 
+	// "If multiple Host headers or multiple :authority headers are present, the
+	// request must be rejected with an HTTP status code 400 as required by Host
+	// validation in RFC 7230 §5.4, gRPC status code INTERNAL, or RST_STREAM
+	// with HTTP/2 error code PROTOCOL_ERROR." - A41. Since this is a HTTP/2
+	// error, this takes precedence over a client not speaking gRPC.
+	if len(mdata[":authority"]) > 1 || len(mdata["host"]) > 1 {
```

Comment:
What if `:authority` (or `host`) is set twice, but one of them is `""`?  Should we ignore those or is that still an error?

Reply:
I think we should still consider that an error. That seems like the callers fault, and a super edge casey thing to add to the server side transport.